### ISSUE TITLE
Add Fshare.vn Storage Provider

### ISSUE THREAD

**Issue by sntran** [rclone/rclone#5009]
#### What is your current rclone version (output from `rclone version`)?

```
rclone v1.53.4
- os/arch: darwin/amd64
- go version: go1.15.6
```

#### What problem are you are trying to solve?

There is currently no support for the provider fshare.vn

#### How do you think rclone should be changed to solve that?

`rclone` can provide a new remote type for fshare.vn to manage files and mount the cloud drive.

They have recently released their API at https://www.fshare.vn/api-doc

---

On behalf of a lot of Vietnamese Internet users, I would like to request support for this provider, now that there is an official API. Understandably, there are still part of the API document that has not been translated to English, but I am available to answer any questions you may have about the API.

I **know** for a fact that there is a lot of interest for this, as Fshare is the biggest and most used storage provider in Vietnam. I can ask the community over there to comment on this issue to show interest, but I think it will just derail the issue at hand.

Thanks for taking your time!

**Comment by ncw** [comment]
Is it possible to get a free account for testing?

**Comment by sntran** [comment]
> Is it possible to get a free account for testing?

Yes, absolutely. I can register one for you if you need. Is it okay for me to send the test account information to your email address on file here?

**Comment by streetsamurai00mi** [comment]
I can send a VIP (1-5days) account if needed

**Comment by vominhtrithcsdhtp1** [comment]
I really hope that happens

**Comment by thip2618** [comment]
hey! Rclone Team please respond to users in Vietnam. We look forward to this coming true.

**Comment by ncw** [comment]
Are there any Vietnamese coders who would like to take on the challenge of writing this backend?

**Comment by sntran** [comment]
Hi Nick,

I know at least one developer was working on this. I'll check with them again.

If nobody else comes forward, I'll take a stab at it after June.

**Comment by ncw** [comment]
That would be great @sntran  - thank you :-)

**Comment by sntran** [comment]
Is there a general documentation for implementing a new backend? Which functions should I implement?

Or should I just look at other PRs to figure it out?

**Comment by ncw** [comment]
Check out the [CONTRIBUTING](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md) guide, and in particular the [writing a new backend section](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#writing-a-new-backend)

Once you've got the skeleton of the backend working, you can use the unit/integration tests to guide your development to get it working. It is actually kind of fun because of that!

Feel free to ask questions if you get stuck.

Note also that I'll need an account for the integration test server (which runs the integration tests daily).

**Comment by oldboy5555** [comment]
> Có tài liệu chung nào hướng dẫn triển khai back-end mới không? Tôi nên triển khai những chức năng nào?
> 
> Hay tôi nên xem các PR khác để tìm ra câu trả lời?

có gì tiến triển mới không bác? tới 2025 rồi mà list host support vẫn ko có Fshare.vn nhỉ

**Comment by ncw** [comment]
@oldboy5555 see https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#writing-a-new-backend - all the docs on how to make a new backend are there.
